### ISSUE TITLE
[Question] is it possible to download m3u8 live video with --downloader "Dash,m3u8:native" ?

### ISSUE THREAD

**Issue by erickyun** [yt-dlp/yt-dlp#262]
## Checklist

- [x] I'm asking a question
- [x] I've looked through the README and FAQ for similar questions
- [x] I've searched the bugtracker for similar questions including closed ones

## Question

When I try to download m3u8 live video it only downloads 3 segments and then it stops downloading?

Can I download live m3u8 video with --downloader "Dash,m3u8:native" or can you add this feature for live m3u8 video if it is not supported?

Command I use:

```
yt-dlp 'https://mn-nl.mncdn.com:443/blutv_trt1/smil:trt1_sd.smil/chunklist_b2064000.m3u8' --fragment-retries 'infinite' -f 'best' -o 'liveVid.%(ext)s' --downloader "Dash,m3u8:native"
```

**Comment by pukkandan** [comment]
No, live videos are not supported by the native downloader. It will be handed over to ffmpeg

**Comment by erickyun** [comment]
> No, live videos are not supported by the native downloader. It will (normally) be handed over to ffmpeg.
 
But with ffmpeg segments will be skipped if there a problem occurs right? İt would be nice if 
`--fragment-retries 'infinite'` could also work with ffmpeg then? 

When you have time for it. I would be appreciated if you could add this feature. if it is possible?

**Comment by pukkandan** [comment]
Yes, it is technically possible, but also quite difficult to implement

**Comment by pukkandan** [comment]
> But with ffmpeg segments will be skipped if there a problem occurs right? İt would be nice if
--fragment-retries 'infinite' could also work with ffmpeg then?

ffmpeg doesnt support retrying afaik. So it is not possible

**Comment by erickyun** [comment]
> Yes, it is technically possible, but also quite difficult to implement

Could you maybe think adding this feauture when you have time for it please?

**Comment by 10362227** [comment]
> Yes, it is technically possible, but also quite difficult to implement

https://live.corusdigitaldev.com/groupd/live/49a91e7f-1023-430f-8d66-561055f3d0f7/live.isml/.mpd this is livestream, please

**Comment by erickyun** [comment]
Does yt-dlp now have the ability to download live streams natively or did I found a workaround? 

Cuz with this command on termux I'm able to download live streams?

```
yt-dlp $(youtube-dl -g "URL") --downloader "dash,m3u8:native" -o vid.ts -R infinite --fragment-retries infinite --no-config
```

**Comment by pukkandan** [comment]
Don't think it works. It will just download the stream from start to the present, but wont wait for and download future segments

**Comment by erickyun** [comment]
> Don't think it works. It will just download the stream from start to the present, but wont wait for and download future segments

yes you're right. 
it did just as you said. 
sorry to bother you was excited it might work :(

**Comment by h82b8** [comment]
I would also appreciate if this could get added as a feature since ffmpeg relatively often hangs when capturing livestreams on twitch, it will just hang indefinitely.

it's pretty well explained here: http://michael-peeters.blogspot.com/2019/03/hanging-m3u8-downloads-in-ffmpeg.html?m=1

since from ffmpegs side there seems to be no interest in fixing this or providing a way to set a timeout, the only way to not miss out on hours of recording sometimes is to not use ffmpeg.

all -timeout as well as -reconnect options that ffmpeg has do not work for this issue

**Comment by batbone** [comment]
Is `--downloader 'dash,m3u8:native'` better than using `aria2`?

**Comment by bashonly** [comment]
> Is `--downloader 'dash,m3u8:native'` better than using `aria2`?

@batbone If you use the `-N` option and `--extractor-args "youtube:formats=dashy", then yes, it is better IMO

This is off-topic though
